server/src/main/java/org/apache/druid/server/coordinator/SegmentReplicantLookup.java

```diff
@@ -59,7 +59,10 @@ public static SegmentReplicantLookup make(DruidCluster cluster)
           if (numReplicants == null) {
             numReplicants = 0;
           }
-          loadingSegments.put(segment.getId(), server.getTier(), numReplicants + 1);
+          // Timed out segments need to be replicated in another server for faster availability
+          if (!serverHolder.getPeon().getTimedOutSegments().contains(segment)) {
+            loadingSegments.put(segment.getId(), server.getTier(), numReplicants + 1);
```

Comment:
As @himanshug pointed out in https://github.com/apache/druid/issues/10193#issuecomment-790941778, there could be two types of slow segment loading.

- There are a few historicals being slow in segment loading in the cluster. This can be caused by unbalanced load queues or some intermittent failures.
- Historicals are OK, but ingestion might outpace the ability to load segments.

This particular change in `SegmentReplicantLookup` could help in the former case, but make things worse in the latter case. In an extreme case, all historicals could have the same set of timed-out segments in their load queue. This might be still OK though, because, if that's the case, Druid cannot get out of that state by itself anyway. The system administrator should add more historicals or use more threads for parallel segment loading. However, we should provide relevant data so that system administrators can tell what's happening. I left another comment about emitting exceptions to provide such data.

Reply:
@jihoonson @himanshug Would it make sense to make the replication behavior user configurable?  We could have a dynamic config like `replicateAfterLoadTimeout` which would control whether the segments would be attempted to be replicated to a different historical in case of a load timeout to the current historical. The default could be true but a cluster operator can set this to false if they wish to avoid the additional churn and know the historicals are OK and it would eventually load the segments.

Reply:
Adding a config seems reasonable to me :+1:

Reply:
It sounds good to me too.

Reply:
Added a config. I've set `replicateAfterLoadTimeout` to false as the default I feel it might be better to preserve the existing behaviour and admins need to be aware of this property's behavior before setting it to true. Let me know what you think.

Reply:
It sounds good to me to preserve the existing behavior by default.